nativeunwind/elfunwindinfo/elfehframelookup.go

```diff
@@ -0,0 +1,52 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package elfunwindinfo // import "go.opentelemetry.io/ebpf-profiler/nativeunwind/elfunwindinfo"
+
+import (
+	"errors"
+	"sort"
+
+	"go.opentelemetry.io/ebpf-profiler/libpf"
+	"go.opentelemetry.io/ebpf-profiler/libpf/pfelf"
+)
+
+type FDE struct {
+	PCBegin uintptr
+	PCRange uintptr
+}
+
+// LookupFDE performs a binary search in .eh_frame_hdr for an FDE covering the given addr.
+func LookupFDE(ef *pfelf.File, addr libpf.Address) (FDE, error) {
+	t, err := newEhFrameTable(ef)
```

Comment:
I think `LookupFDE` should be a member of `ehFrameTable`. And make `newEhFrameTable` and `ehFrameTable` public.

This would allow the user to make multiple queries to the table without parsing the headers again.

I know currently there is no use case for it, but this would be symmetric with the way gopclntab API works, and allow for the possibilty in the future. I would not be surprised if some interpreters wanted to chase FDE blocks, or even walk the full FDE table in future. Having the pre-processed table header exposed as `struct` would also allow extending this API more easier in the future.